test/extended/util/nfs.go

```diff
@@ -5,171 +5,65 @@ import (
-								{ContainerPort: 20048, Name: "mountd"},
-								{ContainerPort: 111, Name: "rpcbind"},
-								{ContainerPort: 111, Name: "rpcbind-udp", Protocol: "UDP"},
-							},
-							Resources: kapiv1.ResourceRequirements{
-								Requests: kapiv1.ResourceList{
-									kapiv1.ResourceMemory: resource.MustParse(capacity),
-								},
-							},
-							SecurityContext: &kapiv1.SecurityContext{
-								Privileged: &privileged,
-							},
-						},
-					},
-				},
-			},
-		},
-	}
-}
-
-// CreateNFSServerService creates a service for the nfs replication controller
-func CreateNFSServerService(name string) *kapiv1.Service {
-	return &kapiv1.Service{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Service",
-			APIVersion: "v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   name,
-			Labels: map[string]string{"name": name},
-		},
-		Spec: kapiv1.ServiceSpec{
-			Ports: []kapiv1.ServicePort{
-				{Port: 2049, Name: "nfs"},
-				{Port: 2049, Name: "nfs-udp", Protocol: "UDP"},
-				{Port: 20048, Name: "mountd"},
-				{Port: 111, Name: "rpcbind"},
-				{Port: 111, Name: "rpcbind-udp", Protocol: "UDP"},
-			},
-			Selector: map[string]string{
-				"role": "nfs-server",
-			},
-		},
-	}
-}
-
-// SetupNFSServer sets up an nfs server replication controller with the given capacity
-// and the nfs service
-func SetupNFSServer(oc *CLI, capacity string) (*kapiv1.ReplicationController, *kapiv1.Service, error) {
-	e2e.Logf("Setting up the nfs server")
-	prefix := oc.Namespace()
-	errs := []error{}
-
+// SetupK8SNFSServerAndVolume sets up an nfs server pod with count number of persistent volumes
+func SetupK8SNFSServerAndVolume(oc *CLI, count int) (*kapiv1.Pod, []*kapiv1.PersistentVolume, error) {
 	e2e.Logf("Adding privileged scc from system:serviceaccount:%s:default", oc.Namespace())
-	if _, err := oc.AsAdmin().Run("adm").Args("policy", "add-scc-to-user", "privileged", fmt.Sprintf("system:serviceaccount:%s:default", oc.Namespace())).Output(); err != nil {
+	_, err := oc.AsAdmin().Run("adm").Args("policy", "add-scc-to-user", "privileged", fmt.Sprintf("system:serviceaccount:%s:default", oc.Namespace())).Output()
+	if err != nil {
 		return nil, nil, err
 	}
 
-	e2e.Logf("Setting up the nfs server replication controller")
-	rc, err := oc.AdminKubeClient().Core().ReplicationControllers(oc.Namespace()).Create(CreateNFSServerReplicationController(fmt.Sprintf("%s%s", nfsPrefix, prefix), capacity))
-	if err != nil {
-		e2e.Logf("WARNING: unable to create replication controller %s%s: %v\n", nfsPrefix, prefix, err)
-		errs = append(errs, err)
+	e2e.Logf(fmt.Sprintf("Creating NFS server"))
+	config := e2e.VolumeTestConfig{
+		Namespace: oc.Namespace(),
+		Prefix:    "nfs",
+		// this image is an extension of k8s.gcr.io/volume-nfs:0.8 that adds
+		// additional nfs mounts to allow for openshift extended tests with
+		// replicas and shared state (mongo, postgresql, mysql, etc.)
+		ServerImage:   "docker.io/gmontero/nfs-server:latest",
```

Comment:
link to the git repo where this image is defined.

Reply:
will do